src/table/settings/gui_settings.ini

```diff
@@ -130,12 +130,17 @@ def      = false
 str      = STR_CONFIG_SETTING_SMOOTH_SCROLLING
 strhelp  = STR_CONFIG_SETTING_SMOOTH_SCROLLING_HELPTEXT
 
-[SDTC_BOOL]
-var      = gui.right_mouse_wnd_close
-flags    = SF_NOT_IN_SAVE | SF_NO_NETWORK_SYNC
-def      = false
+; This was originally right_mouse_wnd_close, renamed to change type to uint8
+[SDTC_VAR]
+var      = gui.right_click_wnd_close
+type     = SLE_UINT8
+flags    = SF_NOT_IN_SAVE | SF_NO_NETWORK_SYNC | SF_GUI_DROPDOWN
+def      = RCC_NO_ACTION
+min      = RCC_NO_ACTION
+max      = RCC_YES_EXCEPT_STICKY
 str      = STR_CONFIG_SETTING_RIGHT_MOUSE_WND_CLOSE
 strhelp  = STR_CONFIG_SETTING_RIGHT_MOUSE_WND_CLOSE_HELPTEXT
+strval   = STR_CONFIG_SETTING_RIGHT_MOUSE_WND_CLOSE_NO_ACTION
 cat      = SC_BASIC
```

Comment:
I thought I wrote a comment about this, but it seems either GitHub ate it, or I never pressed the little green button. Anyway, on the chance of repeating myself:

Please change the type to `SDTC_OMANY` and add a `full` entry. See `gui.autosave` for an example. Now you can attach a list to it, that translates the integer to a name and back. That way, we don't see in the configuration:

`right_click_wnd_close = 2`, but instead `right_click_wnd_close = close-except-sticky`. This makes it much more understandable.

We have been doing this for all new enums; there are only a ton of old ones we haven't done yet.